File: medium/sumint.rb
File.readlines(ARGV[0]).each do |l|
    l.strip!
    arr = l.split(',')
    max_sum = 0
    positive_sum = 0
    arr.each do |v|
        positive_sum = [v.to_i + positive_sum, 0].max
        max_sum = [max_sum, positive_sum].max
    end
    puts max_sum
end

File: medium/sumint.py
import sys


if __name__ == '__main__':
    f = open(sys.argv[1])
    for l in f:
        arr = [int(x) for x in l.split(',')]
        max_sum = 0
        positive_sum = 0
        for v in arr:
            # Keep adding integers, disregard if sum goes negative
            positive_sum = max((v + positive_sum, 0))
            max_sum = max((max_sum, positive_sum))
        print max_sum
